src/views/MarketFilter.tsx

```diff
@@ -36,27 +39,53 @@ export const MarketFilter = ({
   hideNewMarketButton?: boolean;
   searchPlaceholderKey?: string;
   compactLayout?: boolean;
+  unlaunchedMarketToggle?: {
+    checked: boolean;
+    onCheckedChange: (checked: boolean) => void;
+  };
 }) => {
   const stringGetter = useStringGetter();
   const navigate = useNavigate();
   const { hasPotentialMarketsData } = usePotentialMarkets();
   const { uiRefresh, pml: showLaunchMarkets } = testFlags;
   const showProposeButton = hasPotentialMarketsData && !hideNewMarketButton;
 
+  const displayUnlaunchedMarketToggle = unlaunchedMarketToggle && (
+    <WithLabel
+      label={stringGetter({ key: STRING_KEYS.SHOW_LAUNCHABLE_MARKETS })}
+      tw="flex flex-row items-center"
+    >
+      <Switch
+        name="show-launchable"
+        checked={unlaunchedMarketToggle.checked}
+        onCheckedChange={unlaunchedMarketToggle.onCheckedChange}
+        tw="font-mini-book"
+      />
+    </WithLabel>
+  );
+
+  const filterLaunchable = (filter: MarketFilters) => {
+    if (unlaunchedMarketToggle?.checked) return true;
+    return filter !== MarketFilters.LAUNCHABLE;
+  };
+
   const filterToggles = (
     <$ToggleGroup
       items={
-        Object.values(filters).map((value) => ({
-          label: stringGetter({ key: MARKET_FILTER_OPTIONS[value].label, fallback: value }),
-          slotAfter: MARKET_FILTER_OPTIONS[value].isNew && (
-            <NewTag>{stringGetter({ key: STRING_KEYS.NEW })}</NewTag>
-          ),
-          value,
-        })) as MenuItem<MarketFilters>[]
+        Object.values(filters)
+          .filter(filterLaunchable)
+          .map((value) => ({
+            label: stringGetter({ key: MARKET_FILTER_OPTIONS[value].label, fallback: value }),
+            slotAfter: MARKET_FILTER_OPTIONS[value].isNew && (
+              <NewTag>{stringGetter({ key: STRING_KEYS.NEW })}</NewTag>
+            ),
+            value,
+          })) as MenuItem<MarketFilters>[]
```

Comment:
can we get rid of this cast and make it a satisfies or provide the `map` return type? (latter seems cleanest)